core/src/main/java/org/apache/calcite/tools/Frameworks.java

```diff
@@ -134,9 +134,14 @@ public static <R> R withPlanner(final PlannerAction<R> action,
       final FrameworkConfig config) {
     return withPrepare(config,
         (cluster, relOptSchema, rootSchema, statement) -> {
+          List<SchemaPlus> defaults = config.getDefaultSchemas();
+          SchemaPlus initialSchema = rootSchema;
+          if (defaults.size() > 0) {
+            // TODO: Add support for multiple default schema.
+            initialSchema = defaults.get(0);
+          }
           final CalciteSchema schema =
-              CalciteSchema.from(
-                  Util.first(config.getDefaultSchema(), rootSchema));
+              CalciteSchema.from(initialSchema);
```

Comment:
Is this used internally in calcite anywhere? I don't think so, but this may cause problems later

Reply:
I don't believe so and from what I can tell this is just used to get the root so it shouldn't be necessary to handle every default.